At least one close event should be triggered

peol opened the pull request:
Fixes #446 

There is no guarantee that the 'closed' event would only be triggered once, depending on the websocket library used. Instead make sure that the API triggers the event _at least_ once.

wennmo commented on test/integration/suspend-resume.spec.js:
```diff
@@ -80,7 +78,7 @@ describe('QIX Suspend/Resume', () => {
       .then(() => session.close())
       .catch(error => session.close().then(() => Promise.reject(error)))
-      .then(() => expect(closed.callCount).to.equal(1));
+      .then(() => expect(closed.callCount >= 1).to.equal(true));
```
Do we know why there were 2 close events?

peol replied:
It's really hard to trigger locally (about 1 in 100 runs for me), but it seems to be related to the `ws` websocket library, there is nothing in the engine responses indicating why, we just seem to get multiple 'close' events on the socket.

peol commented:
So, after fiddling around for four hours tonight, these are my findings... I SSH:d into Circle CI to get a reproducible environment of the error.

Ran `npm run test:integration` each iteration:
* First fail after 125 iterations
* Added various logging, ran 260 iterations with fail
* Removed logging, had several runs in a row that failed after 5-6 iterations
* Noticed that the error code on the websocket was 1006

Error 1006 indicates that the `onError` handler would be triggered—this is probably where we should look closer, the suspend/resume logic and the session closed events perhaps are acting weird when we enter that logic during a `.resume()`. There are checks for close event codes that could be the cause too if it's mismatching.

This script was used for iterations until fail:

```bash
#!/bin/bash
count=1
while npm run test:integration; do
  echo "iteration $count"
  sleep 2
  (( count++ ))
  # magic, : means rerun while command, which in turn only continues if successful
  :
done
```

peol commented:
Merging this fix for now, we'll have to revisit this if we see production issues similar to the test behavior.
